source/features/highest-rated-comment.tsx

```diff
@@ -0,0 +1,67 @@
+import './highest-rated-comment.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import * as icons from '../libs/icons';
+
+function getCount(reaction: HTMLElement): number {
+	return Number(reaction.textContent!.match(/\d+/)![0]);
+}
+
+function init(): false | void {
+	let highest;
+	const likes = select.all('.js-discussion .js-timeline-item:nth-child(n+6) [aria-label*="reacted with thumbs up"]');
+	for (const like of likes) {
+		const count = getCount(like);
+		const dislike = select('[aria-label*="reacted with thumbs down"]', like.parentElement!);
+
+		if (dislike && getCount(dislike) >= count / 2) {
+			continue; // Controversial comment
+		}
+
+		if (!highest || count > highest.count) {
+			highest = {like, count};
+		}
+	}
+
+	if (!highest || highest.count < 10) {
+		return false;
+	}
+
+	const event = highest.like.closest('.js-timeline-item')!;
+	const text = select('.comment-body', event)!.textContent!.substring(0, 100);
+	const avatar = select('.timeline-comment-avatar', event)!.cloneNode(true);
+	const {hash} = select<HTMLAnchorElement>('.timestamp', event)!;
+
+	select('.unminimized-comment', event)!.classList.add('rgh-highest-rated-comment');
+
+	const position = select.all('.js-comment').indexOf(highest.like.closest('.js-comment') as HTMLElement);
```

Comment:
![image](https://user-images.githubusercontent.com/9121359/58789764-4ab5f100-85c5-11e9-84e4-80d63bb62ea8.png)

Tried that one.

Reply:
This should work
```suggestion
	const position = select.all<Element>('.js-comment').indexOf(highest.like.closest('.js-comment'));
```

Maybe it also needs a `!`

Reply:
I tried some positions for ! and couldn't manage to git this way working. Any clue?

Reply:
This should work
```suggestion
	const position = select.all<Element>('.js-comment').indexOf(highest.like.closest('.js-comment')!);
```

Reply:
Yep, it's one of those I tried.

![image](https://user-images.githubusercontent.com/9121359/58817075-47d8f180-8601-11e9-83b5-ffbed3c93f15.png)

Still a no go.

Reply:
Indeed, the problem moved to `select.all` because the generic (`<...>`) needs to "extend" `HTMLElement`, but `Element` is extended _by_ `HTMLElement`

Your solution was fine. 😅